statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -1287,17 +1461,20 @@ def reorder(self, order):
 #-------------------------------------------------------------------------------
 # VAR Diagnostics: Granger-causality, whiteness of residuals, normality, etc.
 
-    def test_causality(self, equation, variables, kind='f', signif=0.05,
-                       verbose=True):
-        """Compute test statistic for null hypothesis of Granger-noncausality,
-        general function to test joint Granger-causality of multiple variables
+    def test_causality(self, causing, kind='f', signif=0.05, verbose=True):
```

Comment:
I haven't looked at the details yet. (the behavior of this methods confused me several years ago.)

specifically: I think we need both causing and caused variables in the method. The remaining variables are conditioned upon.

in general: 
If we can maintain the same signature but return different results or have different keywords, then we raise a deprecation warning in the old use. 
If we cannot support and distinguish old and new cases, then it would be better to deprecate the entire method and come up with a new name for the replacement.